cmake/externals/sentencepieceproject_cmake.patch

```diff
@@ -1,17 +1,35 @@
-﻿diff --git a/CMakeLists.txt b/CMakeLists.txt
-index a3870cb..159cd84 100644
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index a3870cb..1b060d3 100644
 --- a/CMakeLists.txt
 +++ b/CMakeLists.txt
-@@ -63,7 +63,7 @@ if (MSVC)
-   string(REPLACE "/MD" "/MT" CMAKE_CXX_FLAGS_MINSIZEREL     ${CMAKE_CXX_FLAGS_MINSIZEREL})
-   string(REPLACE "/MD" "/MT" CMAKE_CXX_FLAGS_RELEASE        ${CMAKE_CXX_FLAGS_RELEASE})
-   string(REPLACE "/MD" "/MT" CMAKE_CXX_FLAGS_RELWITHDEBINFO ${CMAKE_CXX_FLAGS_RELWITHDEBINFO})
+@@ -27,6 +27,7 @@ option(SPM_TCMALLOC_STATIC "Link static library of TCMALLOC." OFF)
+ option(SPM_NO_THREADLOCAL "Disable thread_local operator" OFF)
+ option(SPM_USE_BUILTIN_PROTOBUF "Use built-in protobuf" ON)
+ option(SPM_USE_EXTERNAL_ABSL "Use external protobuf" OFF)
++option(SPM_ENABLE_MSVC_MT_BUILD, "Use /MT flag in MSVC build" OFF)
+ 
+ set(CMAKE_CXX_STANDARD 11)
+ set(CMAKE_CXX_STANDARD_REQUIRED ON)
+@@ -59,11 +60,13 @@ else()
+ endif()
+ 
+ if (MSVC)
+-  string(REPLACE "/MD" "/MT" CMAKE_CXX_FLAGS_DEBUG          ${CMAKE_CXX_FLAGS_DEBUG})
+-  string(REPLACE "/MD" "/MT" CMAKE_CXX_FLAGS_MINSIZEREL     ${CMAKE_CXX_FLAGS_MINSIZEREL})
+-  string(REPLACE "/MD" "/MT" CMAKE_CXX_FLAGS_RELEASE        ${CMAKE_CXX_FLAGS_RELEASE})
+-  string(REPLACE "/MD" "/MT" CMAKE_CXX_FLAGS_RELWITHDEBINFO ${CMAKE_CXX_FLAGS_RELWITHDEBINFO})
 -  add_definitions("/wd4267 /wd4244 /wd4305 /Zc:strictStrings /utf-8")
++  if (SPM_ENABLE_MSVC_MT_BUILD)
```

Comment:
This is the old way. Now you can utilize the `CMAKE_MSVC_RUNTIME_LIBRARY`  cmake variable. 
Like this:
https://github.com/microsoft/onnxruntime/blob/main/tools/ci_build/build.py#L1049C9-L1049C12

Reply:
Yes, ort-extensions now leverage this variable, check here for MT build https://github.com/microsoft/onnxruntime-extensions/blob/main/docs/development.md#the-c-shared-library